feat: add `stats/incr/nanwmean`

SergeantQuickscoper opened the pull request:
subject: adds nanwmean package to the stats/incr/* namespace
type: feat

## Description 

Adds nanwmean package to the stats/incr/* namespace

> What is the purpose of this pull request?

This pull request:

-   Adds a package which provides a way to compute the mean of a set of numbers while ignoring NaN values. It is made to address [RFC] Issue #5628, and as suggested in the issue, it is based on a thin wrapper around wmean, similar to the relationship between nansum and sum, maintaining API consistency and design. This commit includes appropriate documentation and tests for the new purpose of the package, styles of which are consistent to the stats/incr/* namespace.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5628 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

-   [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

SergeantQuickscoper committed:
feat (stats/incr/nanwmean): adds nanwmean package to the stats/incr/* namespace

This commit adds a package which provides a way to compute the mean of a set of numbers while ignoring NaN values.
It is made to address [RFC] Issue #5628, and as suggested in the issue, it is based on a thin wrapper around wmean, similar
to the relationship between nansum and sum, mainting API consistency and design. This commit includes appropriate documentation and tests
for the new purpose of the package, styles of which are consistent to the stats/incr/* namespace.

Fixes: #5628 [RFC]
Private-ref: https://github.com/stdlib-js/stdlib/issues/5628
Authored-by: Don Chacko <donisepic30@gmail.com>

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanwmean/README.md:
```diff
@@ -0,0 +1,169 @@
+
+# incrnanwmean
+
+> Compute a [weighted arithmetic mean][weighted-arithmetic-mean] incrementally, while ignoring `NaN` values.
```
```suggestion
> Compute a [weighted arithmetic mean][weighted-arithmetic-mean] incrementally, ignoring `NaN` values.
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanwmean/README.md:
````diff
@@ -0,0 +1,169 @@
+## Usage
+
+```javascript
+var incrnanwmean = require( '@stdlib/stats/incr/wmean' );
````
```suggestion
var incrnanwmean = require( '@stdlib/stats/incr/nanwmean' );
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanwmean/README.md:
```diff
@@ -0,0 +1,169 @@
+
+#### incrnanwmean()
+
+Returns an accumulator `function` which incrementally computes a [weighted arithmetic mean][weighted-arithmetic-mean], while ignoring NaN values.
```
```suggestion
Returns an accumulator `function` which incrementally computes a [weighted arithmetic mean][weighted-arithmetic-mean], ignoring NaN values.
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanwmean/README.md:
````diff
@@ -0,0 +1,169 @@
+Returns an accumulator `function` which incrementally computes a [weighted arithmetic mean][weighted-arithmetic-mean], while ignoring NaN values.
+
+```javascript
+var accumulator = incrwmean();
````
```suggestion
var accumulator = incrnanwmean();
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanwmean/README.md:
```diff
@@ -0,0 +1,169 @@
+
+<section class="related">
+
+* * *
```
Remove L140-147

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanwmean/README.md:
```diff
@@ -0,0 +1,169 @@
+- [`@stdlib/stats/incr/wmean`](../wmean): compute an arithmetic mean incrementally.
+- [`@stdlib/stats/incr/nanmean`](../nanmean): compute an arithmetic mean incrementally, while ignoring NaN values.
+
+
```
</section> is also missing for related.

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanwmean/README.md:
```diff
@@ -0,0 +1,169 @@
+
+[weighted-arithmetic-mean]: https://en.wikipedia.org/wiki/Weighted_arithmetic_mean
+
+<!-- <related-links> -->
```
Remove L157-165

SergeantQuickscoper committed:
fixup! feat: feat (stats/incr/nanwmean): adds nanwmean package to the stats/incr/* namespace

This commit adds minor changes to the README.md of the nanwmean package, to improve clarity and fix a mistake in the usage section where a different package was used. 

Author: Don Roy Chacko <donisepic30@gmail.com>
Reviewed-by: hrshya

Co-authored-by: Harsh <149176984+hrshya@users.noreply.github.com>
Signed-off-by: Don Roy Chacko <52739172+SergeantQuickscoper@users.noreply.github.com>

SergeantQuickscoper committed:
Merge branch 'stdlib-js:develop' into feature/nanwmean

SergeantQuickscoper committed:
fixup! feat (stats/incr/nanwmean): fixes lint errors

This commit fixes the lint errors in the nanwmean module. The changes are applied to the following files:
- nanwmean/README.md
- nanwmean/docs/repl.txt
- nanwmean/lib/index.js
- nanwmean/lib/main.js
- nanwmean/test/test.js
Improvements include:
- Correcting indentation and spacing issues
- Switch to LF line endings

SergeantQuickscoper committed:
fixup! feat (stats/incr/nanwmean): fixes remark lint errors

This commit fixes remark lint errors in the README.md file for the `@stdlib/stats/incr/nanwmean` package.

Authored By: Don Roy Chacko <donisepic30@gmail.com>

SergeantQuickscoper committed:
Merge branch 'stdlib-js:develop' into feature/nanwmean

SergeantQuickscoper commented:
/stdlib lint-autofix

stdlib-bot committed:
fix: resolve lint errors

SergeantQuickscoper committed:
fixup! feat (stats/incr/nanwmean): fix final license header lint

This commit is a license header lint fix for the nanwmean module for the README.md file.

SergeantQuickscoper committed:
Merge branch 'feature/nanwmean' of github.com:SergeantQuickscoper/stdlib into feature/nanwmean

This commit accounts for the merge of the lint_autofix commit to the feature/nanwmean branch.

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanwmean/lib/index.js:
```diff
@@ -0,0 +1,54 @@
+* // returns 2.0
+*
+* mean = accumulator( -5.0, 2.0 );
+* // returns -2.0
+*
+* mean = accumulator();
+* // returns -2.0
```
Computed value is wrong. After `(2.0, w=1.0)` then `(-5.0, w=2.0)`, the weighted mean is `(2*1 + (-5)*2) / (1+2) = -8/3 ≈ -2.67`, not `-2.0`. `main.js` JSDoc has the correct value.

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanwmean/docs/types/index.d.ts:
```diff
@@ -0,0 +1,63 @@
+* // returns 2.666...
+*
+* v = accumulator( 4.0, 3.0 );
+* // returns 3.166...
+*
+* v = accumulator();
+* // returns 3.166...
```
Wrong value. After `(2.0, w=1)`, `(3.0, w=2)`, `(4.0, w=3)`, the weighted mean is `(2+6+12)/6 = 20/6 ≈ 3.333`, not `3.166...`.

Same issue on line 56.

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanwmean/benchmark/benchmark.js:
```diff
@@ -0,0 +1,66 @@
+	for ( i = 0; i < b.iterations; i++ ) {
+		v = ( randu() < 0.2 ) ? NaN : randu() * 100.0;
+		w = ( randu() < 0.2 ) ? NaN : randu() * 10.0;
+		acc( v, w );
```
Capture the return value of `acc( v, w )` and add a sanity check both in-loop and after `b.toc()`. See the `stats/incr/wmean` benchmark for reference.

Planeshifter reviewed (changes requested):
Thanks for opening a PR!
